test/test_dynamic_shapes.py

```diff
@@ -1818,6 +1818,95 @@ def test_stride_symnode(self):
         self.assertTrue(isinstance(s3, int))
         self.assertTrue(str(s1.node.expr) != str(s2.node.expr))
 
+    @fresh_cache()
+    @torch._dynamo.config.patch("capture_scalar_outputs", True)
+    @parametrize("backend", ["inductor", "eager"])
+    def test_dynamic_int_basic_compile(self, backend):
+        from torch.fx.experimental.sym_node import DynamicInt
+
+        cnt = CompileCounterWithBackend(backend)
+
+        # test scalar inputs to function
+        @torch.compile(fullgraph=True, backend=cnt)
+        def fn(x, y, z):
+            out = torch.tensor([x + y + z])
+            out = out + torch.zeros(abs(x) + 2).sum()  # test out tensor construction
+            return out
+
+        x = DynamicInt(1)
+        z = DynamicInt(3)
+        self.assertEqual(fn(x, x, z), 5)  # guard: x == y
```

Comment:
instead of hard coding the answer, can you compare with eager?

eg.

fn_c = torch.compile(fn)
self.assertEqual(fn(x, y, z), fn_c(x, y, z))